tests/test_dayobs.py

```diff
@@ -54,3 +66,86 @@ def test_dayobs(self):
                 self.assert_equal(DayObs.from_time(t.mjd), d)
+
+    def test_rs_time(self):
+        num_nights_tested = 1
+
+        rng = np.random.default_rng(seed=6563)
+        test_mjds = rng.choice(
+            np.arange(DayObs.from_date("2024-10-01").mjd, DayObs.from_date("2036-10-01").mjd),
+            num_nights_tested,
+            replace=False,
+        )
+        test_directions = ["rise", "set"]
+        test_alts = [-18, -12, -6, 0]
+        test_bodies = ["sun", "moon"]
+
+        for mjd in test_mjds:
+            day_obs = DayObs.from_date(mjd, int_format="mjd")
+            for body in test_bodies:
+                for direction in test_directions:
+                    for alt in test_alts:
+                        try:
+                            event_time = day_obs.body_time(body, alt, direction)
+                        except ValueError:
+                            # Maybe the event just never happens on night.
+                            # But, the sun has the tested events every night.
+                            assert body != "sun"
+                            continue
+
+                        event_altaz = AltAz(obstime=event_time, location=day_obs.location)
+                        event_alt = get_body(body, event_time).transform_to(event_altaz).alt.deg
+                        assert np.isclose(alt, event_alt, atol=0.01)
+
+                        assert DayObs.from_time(event_time).mjd == day_obs.mjd
+
+                        after_event_time = event_time + 1 * u.minute
+                        after_event_altaz = AltAz(obstime=after_event_time, location=day_obs.location)
+                        after_event_alt = (
+                            get_body(body, after_event_time).transform_to(after_event_altaz).alt.deg
+                        )
+
+                        before_event_time = event_time - 1 * u.minute
+                        before_event_altaz = AltAz(obstime=before_event_time, location=day_obs.location)
+                        before_event_alt = (
+                            get_body(body, before_event_time).transform_to(before_event_altaz).alt.deg
+                        )
+
+                        if direction == "rise":
+                            assert before_event_alt <= alt <= after_event_alt
+                        else:
+                            assert before_event_alt >= alt >= after_event_alt
+
+    @unittest.skipUnless(ALMANAC is not None, "rubin_scheduler almanac not available.")
+    def test_against_almanac(self):
+        num_nights_tested = 2
+        almanac_times = (
+            pd.DataFrame(ALMANAC.sunsets)
+            .sample(num_nights_tested, random_state=42)
+            .set_index("sunset", drop=False)
+        )
+        for sunset_mjd, night_times in almanac_times.iterrows():
+            day_obs = DayObs.from_time(sunset_mjd)
+            assert np.isclose(night_times.sunset, day_obs.sunset.mjd)
```

Comment:
I'm not sure the defaults for np.isclose are set up to catch the kinds of differences we'd like to catch, when being used on values that are on the order of 60980.0 .. 
I could very well be missing something, because this seems crazy but .. 
```
s = SURVEY_START_MJD
s2 = s + 3/24
rtol=1e-05
atol=1e-08
print(s, s2, (np.abs(s - s2)), (atol + rtol * np.abs(s2)), (atol + rtol * np.abs(s2))*24)
np.isclose([s], [s2])
```
indicates that this would match for differences up to 14 hours.

Reply:
I just revised the test to set rtol=0 and atol = 1.0/(24*60), and did some experiments to verify it does what I expect.